ivy/stateful/module.py

```diff
@@ -428,6 +447,76 @@ def found_dup_callback(x, kc):
             vs = vs.cont_prune_key_chain(dup_kc)
         return vs, keychain_mappings
 
+    def _buffer_tracker(self, func):
+        """Tracks the variables defined as buffer variables and stores them."""
+
+        def wrapper(*args, **kwargs):
+            initial_object_snapshot = self.__dict__.copy()
+            # initialise the buffers
+            func(*args, **kwargs)
+            final_object_snapshot = self.__dict__
+            getattr(self, "buffers", {}).update(
+                set(final_object_snapshot.keys()).difference(
+                    set(initial_object_snapshot.keys())
+                )
+            )
+
+        wrapper.buffers_tracked = True
+        return wrapper
+
+    def _get_buffers(self):
```

Comment:
I'm not sure why we need this method actually, given that we are already using `_find_buffers` while building, won't `self.buffers` be a container or arrays anyway?